client/volume_list.go

```diff
@@ -15,7 +16,11 @@ type VolumeListOptions struct {
 
 // VolumeListResult holds the result from the [Client.VolumeList] method.
 type VolumeListResult struct {
-	Items volume.ListResponse
+	// List of volumes.
+	Items []volume.Volume
```

Comment:
FWIW for the client, I changed this to a slice of values instead of pointers. Happy to change to pointers if anyone can come up with good reasons why it should be a pointer 😅 

https://github.com/moby/moby/blob/8222a3f1d93d5ed7fcb9e2dd9ce40521a07859b0/api/types/volume/list_response.go#L13-L16